src/physics/reynolds_averaged_navier_stokes.cpp

```diff
@@ -0,0 +1,823 @@
+                                  const std::array<real,nstate> &conservative_soln2) const
+{
+    std::array<dealii::Tensor<1,dim,real>,nstate> conv_num_split_flux;
+    std::array<real,dim+2> conservative_soln1_rans = extract_rans_conservative_solution(conservative_soln1);
+    std::array<real,dim+2> conservative_soln2_rans = extract_rans_conservative_solution(conservative_soln2);
+
+    const real mean_density = this->navier_stokes_physics->compute_mean_density(conservative_soln1_rans, conservative_soln2_rans);
+    const dealii::Tensor<1,dim,real> mean_velocities = this->navier_stokes_physics->compute_mean_velocities(conservative_soln1_rans,conservative_soln2_rans);
+    const std::array<real,nstate-(dim+2)> mean_turbulence_property = compute_mean_turbulence_property(conservative_soln1, conservative_soln2);
+
+    for (int i=0;i<dim+2;i++){
+        conv_num_split_flux[i] = 0.0;
+    }
+    for (int i=dim+2;i<nstate;i++)
+    {
+        for (int flux_dim = 0; flux_dim < dim; flux_dim++){
+            conv_num_split_flux[i][flux_dim] = mean_density*mean_velocities[flux_dim]*mean_turbulence_property[i-(dim+2)];
+        }
+    }
+
+    return conv_num_split_flux;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+std::array<real,nstate-(dim+2)> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::compute_mean_turbulence_property(const std::array<real,nstate> &conservative_soln1,
+                                   const std::array<real,nstate> &conservative_soln2) const
+{
+    std::array<real,nstate-(dim+2)> mean_turbulence_property;
+
+    const std::array<real,nstate-(dim+2)> primitive_soln1_turbulence_model = convert_conservative_to_primitive_turbulence_model(conservative_soln1); 
+    const std::array<real,nstate-(dim+2)> primitive_soln2_turbulence_model = convert_conservative_to_primitive_turbulence_model(conservative_soln2); 
+
+    for (int i=0;i<nstate-(dim+2);i++){
+        mean_turbulence_property[i] = (primitive_soln1_turbulence_model[i]+primitive_soln2_turbulence_model[i])/2.0;
+    }
+
+    return mean_turbulence_property;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+std::array<real,nstate> ReynoldsAveragedNavierStokesBase<dim,nstate,real>
+::convective_eigenvalues (
+    const std::array<real,nstate> &conservative_soln,
+    const dealii::Tensor<1,dim,real> &normal) const
+{
+    std::array<real,dim+2> conservative_soln_rans = extract_rans_conservative_solution(conservative_soln);
+    const dealii::Tensor<1,dim,real> vel = this->navier_stokes_physics->template compute_velocities<real>(conservative_soln_rans);
+    std::array<real,nstate> eig;
+    real vel_dot_n = 0.0;
+    for (int d=0;d<dim;++d) { vel_dot_n += vel[d]*normal[d]; };
+    for (int i=0; i<dim+2; i++) {
+        eig[i] = 0.0;
+    }
+    for (int i=dim+2; i<nstate; i++) {
+        eig[i] = vel_dot_n;
+    }
```

Comment:
If its the same as `euler.cpp` but for the additional equation, then you could simply do `const double vel_dot_n = this->navier_stokes_physics->convective_eigenvalues(conservative_soln_rans,normal)[1];`

Reply:
fixed